client/app/Streamfinder.jsx

```diff
@@ -52,14 +53,44 @@ class Streamfinder extends React.Component {
       //this is under the impression auth was valid and currentId was sent
       //to this component and updated via component did update.
       // currentId: 10138,
-      user: 'lil timmy'
+      user: 'lilTimmy'
     };
   }
 
-  updateSession(token) {
-    console.log(token);
-    window.localStorage.setItem('sessionToken', 'fakehash');
-    this.setState({ sessionToken: token });
+  updateSession(username) {
+    const sessionToken = btoa(`{"username": "${username}", "date": ${new Date().getTime()}}`);
+    window.localStorage.setItem('sessionToken', sessionToken);
+    this.setState({ sessionToken });
+    return sessionToken;
+  }
+
+  sessionExpired() {
+    if (this.state.sessionToken) {
+      const expiredTime = (new Date().getTime() - JSON.parse(atob(this.state.sessionToken)).date) / 1000;
```

Comment:
As of right now, this errors out and the app will not load. Error: Unexpected token } in JSON at position 0.

As far as I can tell, this is because atob(this.state.sessionToken) is: }©«!

I've tried solving with console logs but am not sure what exactly I'm dealing with.

Reply:
You'll need to navigate to /auth or log out so that your existing session is removed.